docs/en/contributor/Code-Conventions.md

```diff
@@ -123,39 +94,436 @@ public MyClass {
 }
 ```
 
-Here is the convention in Alluxio source code on different log levels:
+### Use Parameterized Logging
+
+If applicable, logging should be parameterized, to provide good performance and consistent syle.
+
+```java
+// Recommended: Parameterized logging
+LOG.debug("Client {} registered with {}", mClient, mHostname);
+
+// Not Recommended: Non-parameterized logging, hard to read and expensive due to String
+// concatenation regardless of if DEBUG is enabled
+LOG.debug("Client " + mClient + " registered with " + mHostname);
+```
+
+### Error Messages Should Be Descriptive
+
+Error messages should include enough detail and context to determine the issue and potential
+solution. This is important for quick diagnosis or fixes of errors.
+
+```java
+// Recommended: error messages with context
+LOG.error("Client {} failed to register to {}. error: {}", mClient, mHostname, exception);
```

Comment:
for error-level logging, we pretty much always want the stack trace, so instead of `error: {}`, we omit a placeholder for the exception and put it as the last argument:
```java
LOG.error("Client {} failed to register to {}", mClient, mHostname, exception);
```

Reply:
Done